File: templates/problems/15_greedy/06_merge_triplets/python/solution.py
def solve_merge_triplets(triplets: list[list[int]], target: list[int]) -> bool:
    """Check if target triplet can be formed. O(n) time, O(1) space."""
    good = [False, False, False]
    for t in triplets:
        if t[0] > target[0] or t[1] > target[1] or t[2] > target[2]:
            continue
        for i in range(3):
            if t[i] == target[i]:
                good[i] = True
    return good[0] and good[1] and good[2]

File: templates/problems/15_greedy/06_merge_triplets/typescript/solution.ts
/**
 * Check if target triplet can be formed. O(n) time, O(1) space.
 */
export function solveMergeTriplets(triplets: number[][], target: number[]): boolean {
    const good = [false, false, false];
    for (const t of triplets) {
        if (t[0] > target[0] || t[1] > target[1] || t[2] > target[2]) continue;
        for (let i = 0; i < 3; i++) {
            if (t[i] === target[i]) good[i] = true;
        }
    }
    return good[0] && good[1] && good[2];
}
